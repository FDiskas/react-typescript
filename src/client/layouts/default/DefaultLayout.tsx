import * as React from 'react';
import { HeaderComponent } from '../../components/header/HeaderComponent';
import { FooterComponent } from '../../components/footer/FooterComponent';

import * as styles from './DefaultLayout.scss';

export interface IDefaultLayoutProps {}

export interface IDefaultLayoutState {}

export class DefaultLayout extends React.Component<IDefaultLayoutProps, IDefaultLayoutState> {
    constructor(props: IDefaultLayoutProps) {
        super(props);
    }

    render() {
        return (
            <div className={styles.layout}>
                Component name: DefaultLayout
                <HeaderComponent key={'header'} />
                <main key={'content'}>{this.props.children}</main>
                <FooterComponent key={'footer'} />
            </div>
        );
    }
}
