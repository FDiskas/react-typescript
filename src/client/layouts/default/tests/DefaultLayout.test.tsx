import * as React from 'react';
import { shallow } from 'enzyme';

import { DefaultLayout } from '../DefaultLayout';

describe('DefaultLayout.js', () => {
    it('renders without crashing', () => {
        shallow(<DefaultLayout/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<DefaultLayout/>);
        expect(wrapper).toMatchSnapshot();
    });
});
