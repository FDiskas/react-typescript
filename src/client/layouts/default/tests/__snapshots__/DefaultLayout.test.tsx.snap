// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout.js matches snapshot 1`] = `
<div
  className="layout"
>
  Component name: DefaultLayout
  <HeaderComponent
    key="header"
  />
  <main
    key="content"
  />
  <FooterComponent
    key="footer"
  />
</div>
`;
